libs/velo-external-db-core/src/converters/data_utils.ts

```diff
@@ -16,3 +18,32 @@ export const generateIdsIfNeeded = (item: Item): ItemWithId => {
 
 const packDates = (item: Item) => Object.entries(item)
                                 .reduce((o, [k, v]) => ({ ...o, [k]: isDate(v) ? { $date: new Date(v).toISOString() } : v }), {})
+
+
+export const domainToSpiErrorTranslator = (err: any): ApplicationError => {
+    switch(err.constructor) {
+        case domainErrors.ItemAlreadyExists: 
+        const itemAlreadyExists: domainErrors.ItemAlreadyExists = err
+        return { code: ApiErrors.WDE0074, description: itemAlreadyExists.message }
+        
+        case domainErrors.CollectionDoesNotExists:
+        const collectionDoesNotExists: domainErrors.CollectionDoesNotExists = err
+        return { code: ApiErrors.WDE0025, description: collectionDoesNotExists.message }
+        
+        case domainErrors.FieldAlreadyExists:
+        const fieldAlreadyExists: domainErrors.FieldAlreadyExists = err
+        return { code: ApiErrors.WDE0074, description: fieldAlreadyExists.message }
+        
+        case domainErrors.FieldDoesNotExist:
+        const fieldDoesNotExist: domainErrors.FieldDoesNotExist = err
+        return { code: ApiErrors.WDE0147, description: fieldDoesNotExist.message }
+    
+        case domainErrors.UnsupportedSchemaOperation:
+        const unsupportedSchemaOperation: domainErrors.UnsupportedSchemaOperation = err
+        return { code: ApiErrors.WDE0025, description: unsupportedSchemaOperation.message }
+        
+        default:
+        return { code: ApiErrors.WDE0112, description: err.message }
+    }
+}
+                                
```

Comment:
we have file called `domaon-to-spi-error-translator`, and there is code very similar to this.
there we are using it with error classes instead of building the object, I think let's stick to that.